### PR TITLE
M2CE-1 Build M2CE docker image

### PR DESCRIPTION
**Pull request by rmustafaev**
bin - custom scripts to operate magento setup, start processes
conf - configuration files for php

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  rm -rf /srv/www && mkdir -p /srv/www && \
+  chmod 775 /srv/www/ && \
+  git clone --branch $M2SETUP_VERSION --depth 1 https://github.com/magento/magento2.git /srv/www && \
+  rm -rf /srv/www/.git* && \
+  chown www-data:www-data -R /srv/www/ && \
+  cd /srv/www/ && \
+  su -c "/usr/local/bin/composer install" -s /bin/sh www-data && \
+  /usr/local/bin/mage-setup && \
+  apt-get purge -y mariadb-server && \
+  rm -rf /var/lib/mysql
```
Comment (marvin4ward) [inline Dockerfile]: consider add `apt-get clean`

File: bin/mage-setup [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+else 
+  echo "Running Magento 2 setup script..."
+  su -c "/usr/local/bin/php -d memory_limit=2G /srv/www/bin/magento setup:install --db-host=$M2SETUP_DB_HOST --db-name=$M2SETUP_DB_NAME --db-user=$M2SETUP_DB_USER --db-password=$M2SETUP_DB_PASSWORD --base-url=$M2SETUP_BASE_URL --admin-firstname=$M2SETUP_ADMIN_FIRSTNAME --admin-lastname=$M2SETUP_ADMIN_LASTNAME --admin-email=$M2SETUP_ADMIN_EMAIL --admin-user=$M2SETUP_ADMIN_USER --admin-password=$M2SETUP_ADMIN_PASSWORD $M2SETUP_USE_SAMPLE_DATA_STRING" -s /bin/sh www-data
+  echo "Removing write permissions from app/etc for security reasons"
+  chmod -R a-w /srv/www/app/etc/
+
+  # Create folder to upload documents to Magento.
+  mkdir -p /srv/www/var/ftp
+  chown -R www-data:www-data /srv/www/var/ftp
```
Comment (marvin4ward) [inline bin/mage-setup]: Second `chown` ?

File: bin/setup-config [hunk 3]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+WWW_DIR=/srv/www
+
+chown www-data:www-data ${WWW_DIR}/bin/magento
+
+cat <<EOF | crontab -u www-data -
+*/1 * * * * flock -w 1 /tmp/cron.flock -c "/usr/local/bin/php ${WWW_DIR}/update/cron.php" >> ${WWW_DIR}/var/log/cronjob.log 2>&1
+*/1 * * * * flock -w 1 /tmp/m2-cron-run.flock -c "/usr/local/bin/php ${WWW_DIR}/bin/magento cron:run" >> ${WWW_DIR}/var/log/cronjob.log 2>&1
+*/1 * * * * flock -w 1 /tmp/m2-setup-cron-run.flock -c "/usr/local/bin/php ${WWW_DIR}/bin/magento setup:cron:run" >> ${WWW_DIR}/var/log/cronjob.log 2>&1
+EOF
```
Comment (marvin4ward) [inline bin/setup-config]: 3 tasks and each every minute? Does it make sense to pace them ? May be add those 3 commands as sequence in one task?

File: bin/setup-config [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+DISABLED_MODULES=$(su -c "${WWW_DIR}/bin/magento module:status" -s /bin/sh www-data | awk '/List of disabled modules/,0' | grep -v 'List of disabled modules')
+
+if [ "$DISABLED_MODULES" = "None" ]; then
+  echo -e "No disabled modules found"
+else 
+  echo -e "Modules compilation started"
+  rm -rf ${WWW_DIR}/var/di
+  rm -rf ${WWW_DIR}/var/generation
+  rm -rf ${WWW_DIR}/pub/static/*
+  su -c "${WWW_DIR}/bin/magento setup:upgrade" -s /bin/sh www-data
```
Comment (marvin4ward) [inline bin/setup-config]: Consider put user name for `su` first.

File: conf/www.conf [hunk 5]
```diff
@@ -0,0 +1,413 @@
+; Start a new pool named 'www'.
```
Comment (marvin4ward) [inline conf/www.conf]: No need for so many comments here. It is useful for manual change after deploy, which we do not plan to do. 
Also more clean config makes it simple to understand what is really configure here.

File: Dockerfile [hunk 6]
```diff
@@ -63,6 +62,7 @@ RUN apt-get update &&  \
   /usr/local/bin/mage-setup && \
   apt-get purge -y mariadb-server && \
   rm -rf /var/lib/mysql
+  apt-get clean
```
Comment (marvin4ward) [inline Dockerfile]: `&& \` ?

File: mage.env [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+M2SETUP_DB_NAME=magento2
+M2SETUP_DB_USER=magento2
+M2SETUP_DB_PASSWORD=magento2
+M2SETUP_BASE_URL=http://myshop.net
+M2SETUP_ADMIN_FIRSTNAME=Admin
+M2SETUP_ADMIN_LASTNAME=User
+M2SETUP_ADMIN_EMAIL=dummy@setronica.com
+M2SETUP_ADMIN_USER=magento2
+M2SETUP_ADMIN_PASSWORD=magento2
+M2SETUP_ADMIN_URI=admin_11111
```
Comment (marvin4ward) [inline mage.env]: EOL missed. consider add it

File: bin/mage-setup (hunk 2 above)
Comment (rmustafaev) [reply bin/mage-setup]: Directory var/ftp will be created after /srv/www/ were chowned to www-data therefore we have to make chown var/ftp/

Comment (rmustafaev) [reply bin/mage-setup]: /srv/www/var/ftp will be created after /srv/www was chowned. And moreover we do it in one RUN therefor no additional layer.

**Pull request closed without merging**